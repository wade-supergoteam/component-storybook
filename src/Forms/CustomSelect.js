// @flow

// $FlowIssue
import get from "lodash.get";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import themeProp from "../utils/theme";
import {
  customSelectBg,
  customSelectBgSize,
  customSelectBorderColor,
  customSelectBorderRadius,
  customSelectBorderWidth,
  customSelectColor,
  customSelectDisabledBg,
  customSelectDisabledColor,
  customSelectFocusBorderColor,
  customSelectFocusBoxShadow,
  customSelectFontSizeLg,
  customSelectFontSizeSm,
  customSelectHeight,
  customSelectHeightLg,
  customSelectHeightSm,
  customSelectIndicator,
  customSelectLineHeight,
  customSelectPaddingX,
  customSelectPaddingY,
  inputBg,
  inputColor,
} from "./default-theme";

const CustomSelect = styled.select`
  display: inline-block;
  width: 100%;
  padding: ${themeProp("customSelectPaddingY", customSelectPaddingY)} 1.75rem
    ${themeProp("customSelectPaddingY", customSelectPaddingY)} ${themeProp("customSelectPaddingX", customSelectPaddingX)};
  line-height: ${themeProp("customSelectLineHeight", customSelectLineHeight)};
  color: ${themeProp("customSelectColor", customSelectColor)};
  vertical-align: middle;
  background: ${themeProp("customSelectBg", customSelectBg)} ${themeProp("customSelectIndicator", customSelectIndicator)}
    no-repeat right ${themeProp("customSelectPaddingX", customSelectPaddingX)} center;
  background-size: ${themeProp("customSelectBgSize", customSelectBgSize)};
  border: ${themeProp("customSelectBorderWidth", customSelectBorderWidth)} solid
    ${themeProp("customSelectBorderColor", customSelectBorderColor)};

  ${({ theme }) =>
    typeof theme.enableRounded === "undefined" || theme.enableRounded
      ? `border-radius: ${get(theme, "customSelectBorderRadius", customSelectBorderRadius)};`
      : "border-radius: 0;"};

  appearance: none;

  &:focus {
    border-color: ${themeProp("customSelectFocusBorderColor", customSelectFocusBorderColor)};
    outline: 0;
    box-shadow: ${themeProp("customSelectFocusBoxShadow", customSelectFocusBoxShadow)};

    &::-ms-value {
      color: ${themeProp("inputColor", inputColor)};
      background-color: ${themeProp("inputBg", inputBg)};
    }
  }

  &[multiple],
  &[size]:not([size="1"]) {
    height: auto;
    padding-right: ${themeProp("customSelectPaddingX", customSelectPaddingX)};
    background-image: none;
  }

  &:disabled {
    color: ${themeProp("customSelectDisabledColor", customSelectDisabledColor)};
    background-color: ${themeProp("customSelectDisabledBg", customSelectDisabledBg)};
  }

  // Hides the default caret in IE11
  &::-ms-expand {
    opacity: 0;
  }

  ${({ size, theme }) => {
    switch (size) {
      case "small":
        return css`
          height: ${get(theme, "customSelectHeightSm", customSelectHeightSm)};
          font-size: ${get(theme, "customSelectFontSizeSm", customSelectFontSizeSm)};
        `;
      case "large":
        return css`
          height: ${get(theme, "customSelectHeightLg", customSelectHeightLg)};
          font-size: ${get(theme, "customSelectFontSizeLg", customSelectFontSizeLg)};
        `;
      default:
        return css`
          height: ${get(theme, "customSelectHeight", customSelectHeight)};
        `;
    }
  }};
`;

CustomSelect.defaultProps = {
  size: "normal",
};

CustomSelect.defaultProps = {
  size: PropTypes.oneOf(["normal", "small", "large"]),
};

export default CustomSelect;
