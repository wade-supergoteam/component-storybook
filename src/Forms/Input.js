// @flow

import PropTypes from "prop-types";
import styled, { css } from "styled-components";

// $FlowIssue
import get from "lodash.get";

import themeProp from "../utils/theme";

import { borderRadius } from "../utils/border-radius";

import { transition } from "../utils/transition";
import { formControlFocus } from "../utils/forms";

import {
  inputPaddingY,
  inputPaddingX,
  fontSizeBase,
  fontSizeSm,
  fontSizeLg,
  inputLineHeight,
  inputPaddingYsm,
  inputPaddingXsm,
  inputLineHeightSm,
  inputPaddingYlg,
  inputPaddingXlg,
  inputLineHeightLg,
  inputBg,
  inputDisabledBg,
  inputColor,
  inputBorderColor,
  inputBorderWidth,
  inputBoxShadow,
  inputBorderRadius,
  inputBorderRadiusLg,
  inputBorderRadiusSm,
  inputPlaceholderColor,
  inputTransition,
  inputFocusColor,
  inputFocusBg,
  inputFocusBorderColor,
  inputFocusBoxShadow,
} from "./default-theme";

const Input = styled.input`
  display: block;
  width: 100%;
  padding: ${themeProp("inputPaddingY", inputPaddingY)}
    ${themeProp("inputPaddingX", inputPaddingX)};
  font-size: ${themeProp("fontSizeBase", fontSizeBase)};
  line-height: ${themeProp("inputLineHeight", inputLineHeight)};
  color: ${themeProp("inputColor", inputColor)};
  background-color: ${themeProp("inputBg", inputBg)};
  background-image: none;
  background-clip: padding-box;
  border: ${themeProp("inputBorderWidth", inputBorderWidth)} solid
    ${themeProp("inputBorderColor", inputBorderColor)};

  ${({ theme }) =>
    typeof theme.enableRounded === "undefined" || theme.enableRounded
      ? `border-radius: ${get(theme, "inputBorderRadius", inputBorderRadius)};`
      : "border-radius: 0;"}

  ${({ theme }) =>
    (typeof theme.enableShadows === "undefined" || theme.enableShadows) &&
    `box-shadow: ${get(theme, "inputBoxShadow", inputBoxShadow)};`};

  ${transition(themeProp("inputTransition", inputTransition))}

  &::-ms-expand {
    background-color: transparent;
    border: 0;
  }

  ${formControlFocus(
    themeProp("inputFocusColor", inputFocusColor),
    themeProp("inputFocusBg", inputFocusBg),
    themeProp("inputFocusBorderColor", inputFocusBorderColor),
  )};

  ${({ theme }) => css`
    &:focus {
      ${typeof theme.enableShadows === "undefined" || theme.enableShadows
        ? `box-shadow: ${get(theme, "inputBoxShadow", inputBoxShadow)}, ${get(
            theme,
            "inputFocusBoxShadow",
            inputFocusBoxShadow,
          )};`
        : `box-shadow: ${get(theme, "inputFocusBoxShadow", inputFocusBoxShadow)};`};
    }
  `};

  &::placeholder {
    color: ${themeProp("inputPlaceholderColor", inputPlaceholderColor)};
    opacity: 1;
  }

  ${props =>
    props.disabled &&
    css`
      background-color: ${themeProp("inputDisabledBg", inputDisabledBg)};
      opacity: 1;
    `}

  ${({ size, theme }) =>
    size === "small" &&
    css`
      padding: ${themeProp("inputPaddingYsm", inputPaddingYsm)} ${themeProp("inputPaddingXsm", inputPaddingXsm)};
      font-size: ${themeProp("fontSizeSm", fontSizeSm)};
      line-height: ${themeProp("inputLineHeightSm", inputLineHeightSm)};
      ${(typeof theme.enableRounded === "undefined" || theme.enableRounded) &&
        borderRadius(themeProp("inputBorderRadiusSm", inputBorderRadiusSm))};
    `}

  ${({ size, theme }) =>
    size === "large" &&
    css`
      padding: ${themeProp("inputPaddingYlg", inputPaddingYlg)} ${themeProp("inputPaddingXlg", inputPaddingXlg)};
      font-size: ${themeProp("fontSizeLg", fontSizeLg)};
      line-height: ${themeProp("inputLineHeightLg", inputLineHeightLg)};
      ${(typeof theme.enableRounded === "undefined" || theme.enableRounded) &&
        borderRadius(themeProp("inputBorderRadiusLg", inputBorderRadiusLg))};
    `};
`;

Input.defaultProps = {
  size: "normal",
};

Input.defaultProps = {
  size: PropTypes.oneOf(["normal", "small", "large"]),
};

export default Input;
