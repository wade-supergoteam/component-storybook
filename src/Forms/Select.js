import React, { Component } from 'react'
import ReactSelect from 'react-select'
import PropTypes from "prop-types"
import makeAnimated from 'react-select/animated'

const animatedComponents = makeAnimated();

const Select = () => {
  return (
    <ReactSelect />  
  );
};

Select.defaultProps = {
    closeMenuOnSelect: false,
    components: animatedComponents,
    name: "color"
};

Select.propTypes = {
    closeMenuOnSelect: PropTypes.oneOf(["true", "false"]),
    options: PropTypes.object,
    isMulti: PropTypes.boolean,
    name: PropTypes.any

};
export default Select;
