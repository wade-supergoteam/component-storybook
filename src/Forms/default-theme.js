// @flow

export const fontSizeBase = "1rem";
export const fontSizeLg = "1.25rem";
export const fontSizeSm = ".875rem";

export const inputPaddingY = ".375rem";
export const inputPaddingX = ".75rem";
export const inputLineHeight = "1.5";
export const inputPaddingYsm = ".25rem";
export const inputPaddingXsm = ".5rem";
export const inputLineHeightSm = "1.5";
export const inputPaddingYlg = ".5rem";
export const inputPaddingXlg = "1rem";
export const inputLineHeightLg = "1.5";
export const inputBg = "#fff";
export const inputDisabledBg = "#e9ecef";
export const inputColor = "#495057";
export const inputBorderColor = "#ced4da";
export const inputBorderWidth = "1px";
export const inputBoxShadow = "inset 0 1px 1px rgba(0,0,0,0.075)";
export const inputBorderRadius = ".25rem";
export const inputBorderRadiusLg = ".3rem";
export const inputBorderRadiusSm = ".2rem";
export const inputFocusBg = "#fff";
export const inputFocusBorderColor = "#80bdff";
export const inputFocusColor = "#495057";
export const inputFocusWidth = ".2rem";
export const inputFocusBoxShadow = "0 0 0 .2rem rgba(0,123,255,0.25)";
export const inputPlaceholderColor = "#6c757d";
export const inputHeightBorder = "2px";
export const inputHeightInner = "2.25rem";
export const inputHeight = "calc(2.25rem + 2px)";
export const inputHeightInnerSm = "1.8125rem";
export const inputHeightSm = "calc(1.8125rem + 2px)";
export const inputHeightInnerLg = "2.875rem";
export const inputHeightLg = "calc(2.875rem + 2px)";
export const inputTransition = "borderColor .15s ease-in-out,boxShadow .15s ease-in-out";

export const formTextMarginTop = ".25rem";
export const formCheckInputGutter = "1.25rem";
export const formCheckInputMarginY = ".3rem";
export const formCheckInputMarginX = ".25rem";
export const formCheckInlineMarginX = ".75rem";
export const formCheckInlineInputMarginX = ".3125rem";
export const formGroupMarginBottom = "1rem";
export const inputGroupAddonColor = "#495057";
export const inputGroupAddonBg = "#e9ecef";
export const inputGroupAddonBorderColor = "#ced4da";

export const customControlGutter = "1.5rem";
export const customControlSpacerX = "1rem";
export const customControlIndicatorSize = "1rem";
export const customControlIndicatorBg = "#dee2e6";
export const customControlIndicatorBgSize = "50% 50%";
export const customControlIndicatorBoxShadow = "inset 0 .25rem .25rem rgba(0,0,0,0.1)";
export const customControlIndicatorDisabledBg = "#e9ecef";
export const customControlLabelDisabledColor = "#6c757d";
export const customControlIndicatorCheckedColor = "#fff";
export const customControlIndicatorCheckedBg = "#007bff";
export const customControlIndicatorCheckedDisabledBg = "rgba(0,123,255,0.5)";
export const customControlIndicatorCheckedBoxShadow = "none";
export const customControlIndicatorFocusBoxShadow = "0 0 0 1px #fff,0 0 0 .2rem rgba(0,123,255,0.25)";
export const customControlIndicatorActiveColor = "#fff";
export const customControlIndicatorActiveBg = "#b3d7ff";
export const customControlIndicatorActiveBoxShadow = "none";
export const customCheckboxIndicatorBorderRadius = ".25rem";
export const customCheckboxIndicatorIndeterminateBg = "#007bff";
export const customCheckboxIndicatorIndeterminateColor = "#fff";
export const customCheckboxIndicatorIndeterminateBoxShadow = "none";
export const customRadioIndicatorBorderRadius = "50%";
export const customSelectPaddingY = ".375rem";
export const customSelectPaddingX = ".75rem";
export const customSelectHeight = "calc(2.25rem + 2px)";
export const customSelectIndicatorPadding = "1rem";
export const customSelectLineHeight = "1.5";
export const customSelectColor = "#495057";
export const customSelectDisabledColor = "#6c757d";
export const customSelectBg = "#fff";
export const customSelectDisabledBg = "#e9ecef";
export const customSelectBgSize = "8px 10px";
export const customSelectIndicatorColor = "#343a40";
export const customSelectBorderWidth = "1px";
export const customSelectBorderColor = "#ced4da";
export const customSelectBorderRadius = ".25rem";
export const customSelectFocusBorderColor = "#80bdff";
export const customSelectFocusBoxShadow = "inset 0 1px 2px rgba(0,0,0,0.075),0 0 5px rgba(128,189,255,0.5)";
export const customSelectFontSizeSm = "75%";
export const customSelectHeightSm = "calc(1.8125rem + 2px)";
export const customSelectFontSizeLg = "125%";
export const customSelectHeightLg = "calc(2.875rem + 2px)";
export const customFileHeight = "calc(2.25rem + 2px)";
export const customFileFocusBorderColor = "#80bdff";
export const customFileFocusBoxShadow = "0 0 0 .2rem rgba(0,123,255,0.25)";
export const customFilePaddingY = ".375rem";
export const customFilePaddingX = ".75rem";
export const customFileLineHeight = "1.5";
export const customFileColor = "#495057";
export const customFileBg = "#fff";
export const customFileBorderWidth = "1px";
export const customFileBorderColor = "#ced4da";
export const customFileBorderRadius = ".25rem";
export const customFileBoxShadow = "inset 0 1px 1px rgba(0,0,0,0.075)";
export const customFileButtonColor = "#495057";
export const customFileButtonBg = "#e9ecef";

export const formFeedbackMarginTop = ".25rem";
export const formFeedbackFontSize = "80%";
export const formFeedbackValidColor = "#28a745";
export const formFeedbackInvalidColor = "#dc3545";

export const customSelectIndicator = `url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3E%3Cpath fill='%23${customSelectIndicatorColor.slice(
  1,
)}' d='M2 0L0 2h4zm0 5L0 3h4z'/%3E%3C/svg%3E")`;
