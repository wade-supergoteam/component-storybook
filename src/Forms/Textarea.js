import Input from "./Input";

const Textarea = Input.withComponent("textarea");

export default Textarea;
