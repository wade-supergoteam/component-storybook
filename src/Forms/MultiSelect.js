import React, { Component } from 'react'
import ReactSelect from 'react-select'
import PropTypes from "prop-types"
import Select from './Select'

const MultiSelect = ({options}) => {
    return (
        <Select closeMenuOnSelect={false}
                     isMulti
                     options={options}
                     defaultValue={options[0]}
        />
    );
};

Select.propTypes = {
    options: PropTypes.object
};
export default MultiSelect;
