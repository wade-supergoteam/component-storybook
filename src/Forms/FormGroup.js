// @flow

import styled from "styled-components";

import themeProp from "../utils/theme";

import { formGroupMarginBottom } from "./default-theme";

const FormGroup = styled.div`
  margin-bottom: ${themeProp("formGroupMarginBottom", formGroupMarginBottom)};
`;

export default FormGroup;
