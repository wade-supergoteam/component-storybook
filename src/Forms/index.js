import Input from "./Input";
import Select from "./Select";
import MultiSelect from "./MultiSelect";
import Textarea from "./Textarea";
import FormGroup from "./FormGroup";
import CustomSelect from "./CustomSelect";

const Form = { Input, Select, MultiSelect, Textarea, FormGroup, CustomSelect };

export { Input, Select, MultiSelect, Textarea, FormGroup, CustomSelect };
export default Form;
