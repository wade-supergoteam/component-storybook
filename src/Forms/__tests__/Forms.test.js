import React from "react";
import renderer from "react-test-renderer";
import Forms from "../";
import CustomSelect from "../CustomSelect";
import Input from "../Input";
import Select from "../Select";
import Textarea from "../Textarea";
import { ThemeProvider } from "styled-components";



const customInputTheme = {
    enableRounded: false,
    inputFocusColor: "SkyBlue",
    inputFocusBorderColor: "PowderBlue",
    inputFocusBoxShadow: "0 0 0 4px azure",
};


test("Custom Select", () => {
    const wrapper = renderer.create(<CustomSelect />);
    expect(wrapper).toMatchSnapshot();
});

test("Create Input Field", () => {
    const wrapper = renderer.create(<Input />);
    expect(wrapper).toMatchSnapshot();
});

test("Small Input", () => {
    const wrapper = renderer.create(<Input size="small" />);
    expect(wrapper).toMatchSnapshot();
});

test("Large Input", () => {
    const wrapper = renderer.create(<Input size="large" />);
    expect(wrapper).toMatchSnapshot();
});

test("Custom Input", () => {
    const wrapper = renderer.create(<ThemeProvider theme={customInputTheme} />);
    expect(wrapper).toMatchSnapshot();
});

test("Create Select", () => {
    const wrapper = renderer.create(<Select />);
    expect(wrapper).toMatchSnapshot();
});

test("Create Large Select", () => {
    const wrapper = renderer.create(<Select size="large" />);
    expect(wrapper).toMatchSnapshot();
});

test("Create Small Select", () => {
    const wrapper = renderer.create(<Select size="small" />);
    expect(wrapper).toMatchSnapshot();
});

test("Create Textarea", () => {
    const wrapper = renderer.create(<Textarea />);
    expect(wrapper).toMatchSnapshot();
});

