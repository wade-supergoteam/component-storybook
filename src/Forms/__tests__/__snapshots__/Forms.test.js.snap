// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create CustomSelect large 1`] = `
<select
  className="sc-ifAKCX jHapST"
  size="large"
/>
`;

exports[`Create CustomSelect small 1`] = `
<select
  className="sc-ifAKCX ihWeHO"
  size="small"
/>
`;

exports[`Create CustomSelect with enableRounded false 1`] = `
<select
  className="sc-ifAKCX SFHiV"
  size={[Function]}
/>
`;

exports[`Create CustomSelect with rounded borders 1`] = `
<select
  className="sc-ifAKCX lgHZOS"
  size={[Function]}
/>
`;

exports[`Create Input Field 1`] = `
<input
  className="sc-bdVaJa cHRGZp"
  size={[Function]}
/>
`;

exports[`Create Large Select 1`] = `
<input
  className="sc-bdVaJa sc-bwzfXH glnWMt"
  size="large"
/>
`;

exports[`Create Select 1`] = `
<input
  className="sc-bdVaJa sc-bwzfXH dCzVIK"
  size={[Function]}
/>
`;

exports[`Create Small Select 1`] = `
<input
  className="sc-bdVaJa sc-bwzfXH kzTRqo"
  size="small"
/>
`;

exports[`Create Textarea 1`] = `
<textarea
  className="sc-htpNat edctCP"
/>
`;

exports[`Custom Input 1`] = `null`;

exports[`Custom Select 1`] = `
<select
  className="sc-ifAKCX lgHZOS"
  size={[Function]}
/>
`;

exports[`Input Field disabled 1`] = `
<input
  className="sc-bdVaJa kMEKHz"
  disabled={true}
  size={[Function]}
/>
`;

exports[`Input Field small enabledRounded false 1`] = `
<input
  className="sc-bdVaJa gSgZv"
  size="small"
/>
`;

exports[`Input with enableRounded 1`] = `
<input
  className="sc-bdVaJa cHRGZp"
  size={[Function]}
/>
`;

exports[`Input with enableRounded false 1`] = `
<input
  className="sc-bdVaJa eExobB"
  size={[Function]}
/>
`;

exports[`Input with enableShadows 1`] = `
<input
  className="sc-bdVaJa cHRGZp"
  size={[Function]}
/>
`;

exports[`Input with enableShadows false 1`] = `
<input
  className="sc-bdVaJa dhrwem"
  size={[Function]}
/>
`;

exports[`Large Input 1`] = `
<input
  className="sc-bdVaJa iGpphR"
  size="large"
/>
`;
